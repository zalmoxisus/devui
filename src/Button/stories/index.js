import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import styled from 'styled-components';
import { withKnobs, text, boolean, select } from '@kadira/storybook-addon-knobs';
import MdFiberManualRecord from 'react-icons/lib/md/fiber-manual-record';
import Button from '../';

export const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;

  .wrapper {
    display: flex;
    width: 100%;
    padding: 5px;
    background-color: ${props => props.theme.base01};
    text-align: center;
  }
`;

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Container>
        <Button
          title={text('Title', 'Hello Tooltip')}
          tooltipPosition={
            select('tooltipPosition', ['top', 'bottom', 'left', 'right',
             'bottom-left', 'bottom-right', 'top-left', 'top-right'])
          }
          big={boolean('Big', true)}
          disabled={boolean('Disabled', false)}
          onClick={action('button clicked')}
        >
          {text('Label', 'Hello Button')}
        </Button>
      </Container>
    )
  )
  .addWithInfo(
    'mark',
    '',
    () => (
      <Container>
        <Button
          title={text('Title', 'Hello Tooltip')}
          tooltipPosition={
            select('tooltipPosition', ['top', 'bottom', 'left', 'right',
             'bottom-left', 'bottom-right', 'top-left', 'top-right'])
          }
          big={boolean('Big', true)}
          disabled={boolean('Disabled', false)}
          onClick={action('button clicked')}
          mark={text('mark', 'base08')}
        >
          <MdFiberManualRecord />
        </Button>
      </Container>
    )
  );
