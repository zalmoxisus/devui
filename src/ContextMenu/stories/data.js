import React from 'react';

/* eslint-disable react/prop-types */
const Component = ({ selected }) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: '100%',
      fontSize: '22px'
    }}
  >
    Selected {selected}
  </div>
);
/* eslint-enable react/prop-types */

const selector = item => ({ selected: item.name });

export const items = [
  {
    name: 'Menu Item 1',
    component: Component,
    selector
  },
  {
    name: 'Menu Item 2',
    component: Component,
    selector
  },
  {
    name: 'Menu Item 3',
    component: Component,
    selector
  }
];
