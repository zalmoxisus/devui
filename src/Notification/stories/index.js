import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import styled from 'styled-components';
import { withKnobs, text, select } from '@kadira/storybook-addon-knobs';
import Notification from '../';

export const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
`;

storiesOf('Notification', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Container>
        <Notification
          type={
            select('type', ['info', 'success', 'warning', 'error'], 'warning')
          }
          onClose={action('notification closed')}
        >
          {text('Message', 'Hello Notification')}
        </Notification>
      </Container>
    )
  );
