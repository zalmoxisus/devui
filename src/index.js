export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs } from './Tabs';
