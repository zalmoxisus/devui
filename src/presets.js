/* eslint-disable global-require */

import 'codemirror/mode/javascript/javascript';
import 'codemirror/addon/fold/foldgutter';
import 'codemirror/addon/fold/foldcode';
import 'codemirror/addon/fold/brace-fold';

if (process.env.NODE_ENV !== 'test') {
  require('../fonts/index.css');
  require('codemirror/lib/codemirror.css');
  require('codemirror/addon/fold/foldgutter.css');
}
