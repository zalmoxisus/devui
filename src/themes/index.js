import styled from 'styled-components';
import invertColors from '../utils/invertColors';

import material from './material';

import * as schemes from 'redux-devtools-themes';
import defaultScheme from 'base16';
schemes.default = defaultScheme;

export const listSchemes = () => Object.keys(schemes).slice(1);

const themes = {
  material
};

export const listThemes = () => ['default', Object.keys(themes)];

export const getTheme = ({ theme: type, scheme, invert }) => {
  let theme = {
    type,
    light: invert,
    fontFamily: '\'Source Sans Pro\', sans-serif',
    codeFontFamily: '\'Source Code Pro\', monospace',
    inputHeight: 30,
    inputBorderWidth: 1,
    inputBorderRadius: 4
  };
  let colors = schemes[scheme];
  if (invert) colors = invertColors(colors);
  if (type !== 'default') {
    theme = {
      ...theme,
      ...themes[type](colors)
    };
  }
  return {
    ...colors,
    spinnerSize: Math.floor(theme.inputHeight / 2) - 2,
    inputPadding: theme.inputHeight / 3,
    selectArrowWidth: Math.floor(theme.inputHeight / 7),
    inputInternalHeight: theme.inputHeight - theme.inputBorderWidth * 2,
    inputBorderColor: colors.base05,
    inputFocusedStyle: `border-color: ${colors.base0D}`,
    ...theme
  };
};

export const getStyles = (styles, component, multiple) => styled(component || 'div')('',
  props => (
    !multiple ? styles :
      styles[props.theme.type] || styles.default
  )(props)
);
/*
 Equivalent to
 const SelectContainer = styled(ReactSelect)`
   ${props => styles[props.theme.type](props)}
 `;
*/
