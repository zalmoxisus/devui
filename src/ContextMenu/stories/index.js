import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import styled from 'styled-components';
import { withKnobs, text, boolean, select } from '@kadira/storybook-addon-knobs';
import MdFiberManualRecord from 'react-icons/lib/md/fiber-manual-record';
import ContextMenuTrigger from '../';
import { items } from './data';

export const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;

  .wrapper {
    display: flex;
    width: 100%;
    padding: 5px;
    background-color: ${props => props.theme.base01};
    text-align: center;
  }
`;

storiesOf('ContextMenu', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Container>
        <ContextMenuTrigger
          onClick={action('menu item clicked')}
          leftClick={boolean('leftClick', true)}
          rightClick={boolean('rightClick', true)}
          items={items}
        >
          {text('Label', 'ContextMenu')}
        </ContextMenuTrigger>
      </Container>
    )
  );
