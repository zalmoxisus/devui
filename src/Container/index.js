import React, { PropTypes } from 'react';
import { ThemeProvider } from 'styled-components';
import { getTheme } from '../themes';
import { ContainerWrapper } from './styles';

const Container = ({ themeData, className, children }) => (
  <ThemeProvider theme={getTheme(themeData)}>
    <ContainerWrapper className={className}>
      {children}
    </ContainerWrapper>
  </ThemeProvider>
);

Container.propTypes = {
  children: PropTypes.node.isRequired,
  themeData: PropTypes.object.isRequired,
  className: PropTypes.string
};

export default Container;
