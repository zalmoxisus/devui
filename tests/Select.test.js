import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson, mountToJson } from 'enzyme-to-json';
import { Select } from '../src';
import { options } from '../src/Select/stories/options';

describe('Select', function () {
  it('renders correctly', () => {
    const wrapper = render(<Select options={options} onChange={() => {}} />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = render(
      <Select
        options={options}
        onChange={() => {}}
        value="one"
        menuMaxHeight={20}
        clearable
        disabled
        isLoading
        multi
        searchable={false}
        openOuterUp
      />
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('should select another option', () => {
    const onChange = jest.fn();
    const wrapper = mount(<Select options={options} onChange={onChange} />);

    const input = wrapper.find('input');
    input.at(0).instance().value = 'two';
    input.first().simulate('change');
    expect(mountToJson(wrapper)).toMatchSnapshot();
    input.first().simulate('keyDown', { keyCode: 13 });
    expect(onChange).toBeCalled();
  });

  it("shouldn't find any results", () => {
    const onChange = jest.fn();
    const wrapper = mount(<Select options={options} onChange={onChange} />);

    const input = wrapper.find('input');
    input.at(0).instance().value = 'text';
    input.first().simulate('change');
    expect(mountToJson(wrapper)).toMatchSnapshot(); // 'No results found'
    input.first().simulate('keyDown', { keyCode: 13 });
    expect(onChange).not.toBeCalled();
  });
});
