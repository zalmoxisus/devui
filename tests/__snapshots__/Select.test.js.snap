// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  class="Select sc-bZQynM lpCNM Select--single is-searchable"
>
  <div
    class="Select-control"
  >
    <span
      class="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        class="Select-placeholder"
      >
        Select...
      </div>
      <div
        class="Select-input"
        style="display:inline-block"
      >
        <input
          aria-activedescendant="react-select-2--value"
          aria-expanded="false"
          aria-haspopup="false"
          aria-owns=""
          role="combobox"
          style="width:5px;box-sizing:content-box"
          value=""
        />
        <div
          style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
        />
      </div>
    </span>
    <span
      class="Select-arrow-zone"
    >
      <span
        class="Select-arrow"
      />
    </span>
  </div>
</div>
`;

exports[`Select renders with props 1`] = `
<div
  class="Select sc-bZQynM dVPEwd Select--multi is-clearable is-disabled is-loading has-value"
>
  <div
    class="Select-control"
  >
    <span
      class="Select-multi-value-wrapper"
      id="react-select-3--value"
    >
      <div
        class="Select-value"
      >
        <span
          aria-selected="true"
          class="Select-value-label"
          id="react-select-3--value-0"
          role="option"
        >
          One
          <span
            class="Select-aria-only"
          >
            Â 
          </span>
        </span>
      </div>
      <div
        aria-activedescendant="react-select-3--value"
        aria-expanded="false"
        aria-owns=""
        aria-readonly="true"
        class="Select-input"
        role="combobox"
        style="border:0;width:1px;display:inline-block"
        tabindex="0"
      />
    </span>
    <span
      aria-hidden="true"
      class="Select-loading-zone"
    >
      <span
        class="Select-loading"
      />
    </span>
    <span
      class="Select-arrow-zone"
    >
      <span
        class="Select-arrow"
      />
    </span>
  </div>
</div>
`;

exports[`Select should select another option 1`] = `
<Select
  autosize={true}
  clearable={false}
  menuMaxHeight={200}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "One hundred",
        "value": "hundred",
      },
    ]
  }
  simpleValue={true}
>
  <Styled(Select)
    autosize={true}
    clearable={false}
    menuMaxHeight={200}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "One",
          "value": "one",
        },
        Object {
          "label": "Two",
          "value": "two",
        },
        Object {
          "label": "One hundred",
          "value": "hundred",
        },
      ]
    }
    simpleValue={true}
  >
    <Select
      addLabelText="Add \\"{label}\\"?"
      arrowRenderer={[Function]}
      autosize={true}
      backspaceRemoves={true}
      backspaceToRemoveMessage="Press backspace to remove {label}"
      className="sc-bZQynM lpCNM"
      clearAllText="Clear all"
      clearRenderer={[Function]}
      clearValueText="Clear value"
      clearable={false}
      closeOnSelect={true}
      deleteRemoves={true}
      delimiter=","
      disabled={false}
      escapeClearsValue={true}
      filterOptions={[Function]}
      ignoreAccents={true}
      ignoreCase={true}
      inputProps={Object {}}
      isLoading={false}
      joinValues={false}
      labelKey="label"
      matchPos="any"
      matchProp="any"
      menuBuffer={0}
      menuMaxHeight={200}
      menuRenderer={[Function]}
      multi={false}
      noResultsText="No results found"
      onBlurResetsInput={true}
      onChange={[Function]}
      onCloseResetsInput={true}
      onSelectResetsInput={true}
      openOnClick={true}
      optionComponent={[Function]}
      options={
        Array [
          Object {
            "label": "One",
            "value": "one",
          },
          Object {
            "label": "Two",
            "value": "two",
          },
          Object {
            "label": "One hundred",
            "value": "hundred",
          },
        ]
      }
      pageSize={5}
      placeholder="Select..."
      required={false}
      scrollMenuIntoView={true}
      searchable={true}
      simpleValue={true}
      tabSelectsValue={true}
      valueComponent={[Function]}
      valueKey="value"
    >
      <div
        className="Select sc-bZQynM lpCNM Select--single is-open is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-4--value"
          >
            <AutosizeInput
              aria-activedescendant="react-select-4--option-0"
              aria-describedby={undefined}
              aria-expanded="true"
              aria-haspopup="true"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns="react-select-4--list"
              className="Select-input"
              minWidth="5"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              tabIndex={undefined}
              value="two"
            >
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <style
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "input#undefined::-ms-clear {display: none;}",
                    }
                  }
                />
                <input
                  aria-activedescendant="react-select-4--option-0"
                  aria-describedby={undefined}
                  aria-expanded="true"
                  aria-haspopup="true"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns="react-select-4--list"
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value="two"
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  two
                </div>
              </div>
            </AutosizeInput>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
        <div
          className="Select-menu-outer"
          style={undefined}
        >
          <div
            className="Select-menu"
            id="react-select-4--list"
            onMouseDown={[Function]}
            onScroll={[Function]}
            role="listbox"
            style={undefined}
            tabIndex={-1}
          >
            <Option
              className="Select-option is-focused"
              instancePrefix="react-select-4-"
              isDisabled={undefined}
              isFocused={true}
              isSelected={false}
              key="option-0-two"
              onFocus={[Function]}
              onSelect={[Function]}
              option={
                Object {
                  "label": "Two",
                  "value": "two",
                }
              }
              optionIndex={0}
            >
              <div
                className="Select-option is-focused"
                id="react-select-4--option-0"
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseMove={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="option"
                style={undefined}
                title={undefined}
              >
                Two
              </div>
            </Option>
          </div>
        </div>
      </div>
    </Select>
  </Styled(Select)>
</Select>
`;

exports[`Select shouldn't find any results 1`] = `
<Select
  autosize={true}
  clearable={false}
  menuMaxHeight={200}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "One hundred",
        "value": "hundred",
      },
    ]
  }
  simpleValue={true}
>
  <Styled(Select)
    autosize={true}
    clearable={false}
    menuMaxHeight={200}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "One",
          "value": "one",
        },
        Object {
          "label": "Two",
          "value": "two",
        },
        Object {
          "label": "One hundred",
          "value": "hundred",
        },
      ]
    }
    simpleValue={true}
  >
    <Select
      addLabelText="Add \\"{label}\\"?"
      arrowRenderer={[Function]}
      autosize={true}
      backspaceRemoves={true}
      backspaceToRemoveMessage="Press backspace to remove {label}"
      className="sc-bZQynM lpCNM"
      clearAllText="Clear all"
      clearRenderer={[Function]}
      clearValueText="Clear value"
      clearable={false}
      closeOnSelect={true}
      deleteRemoves={true}
      delimiter=","
      disabled={false}
      escapeClearsValue={true}
      filterOptions={[Function]}
      ignoreAccents={true}
      ignoreCase={true}
      inputProps={Object {}}
      isLoading={false}
      joinValues={false}
      labelKey="label"
      matchPos="any"
      matchProp="any"
      menuBuffer={0}
      menuMaxHeight={200}
      menuRenderer={[Function]}
      multi={false}
      noResultsText="No results found"
      onBlurResetsInput={true}
      onChange={[Function]}
      onCloseResetsInput={true}
      onSelectResetsInput={true}
      openOnClick={true}
      optionComponent={[Function]}
      options={
        Array [
          Object {
            "label": "One",
            "value": "one",
          },
          Object {
            "label": "Two",
            "value": "two",
          },
          Object {
            "label": "One hundred",
            "value": "hundred",
          },
        ]
      }
      pageSize={5}
      placeholder="Select..."
      required={false}
      scrollMenuIntoView={true}
      searchable={true}
      simpleValue={true}
      tabSelectsValue={true}
      valueComponent={[Function]}
      valueKey="value"
    >
      <div
        className="Select sc-bZQynM lpCNM Select--single is-open is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-5--value"
          >
            <AutosizeInput
              aria-activedescendant="react-select-5--option-null"
              aria-describedby={undefined}
              aria-expanded="true"
              aria-haspopup="true"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns="react-select-5--list"
              className="Select-input"
              minWidth="5"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              tabIndex={undefined}
              value="text"
            >
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <style
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "input#undefined::-ms-clear {display: none;}",
                    }
                  }
                />
                <input
                  aria-activedescendant="react-select-5--option-null"
                  aria-describedby={undefined}
                  aria-expanded="true"
                  aria-haspopup="true"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns="react-select-5--list"
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value="text"
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  text
                </div>
              </div>
            </AutosizeInput>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
        <div
          className="Select-menu-outer"
          style={undefined}
        >
          <div
            className="Select-menu"
            id="react-select-5--list"
            onMouseDown={[Function]}
            onScroll={[Function]}
            role="listbox"
            style={undefined}
            tabIndex={-1}
          >
            <div
              className="Select-noresults"
            >
              No results found
            </div>
          </div>
        </div>
      </div>
    </Select>
  </Styled(Select)>
</Select>
`;
